spec/controllers/sessions_controller_spec.rb

```diff
@@ -39,17 +39,137 @@
+              post :create, session: {
+                email: @user.email,
+                password: @user.password
+              }
+            end
+          end
+
+          it "sets a custom cookie name in the header" do
+            expect(response.headers["Set-Cookie"]).
+              to match /custom-cookie=old-token/
+          end
+        end
+
+        context "with cookie path set" do
+          before do
+            with_custom_config(:cookie_path, "/cool-page") do
+              post :create, session: {
+                email: @user.email,
+                password: @user.password
+              }
+            end
+          end
+          it "sets a path cookie" do
+            expect(response.headers["Set-Cookie"]).to match /path=\/cool-page/
+          end
+        end
+
+        context "if httponly is set" do
+          before do
+            with_custom_config(:httponly, "true") do
+              post :create, session: {
+                email: @user.email,
+                password: @user.password
+              }
+            end
+          end
+
+          it "sets a httponly cookie" do
+            expect(response.headers["Set-Cookie"]).to match /HttpOnly/
+          end
+        end
+
+        context "if the cookie domain is set" do
+          before do
+            with_custom_config(:cookie_domain, ".example.com") do
+              post :create, session: {
+                email: @user.email,
+                password: @user.password
+              }
+            end
+          end
+
+          it "sets a domain cookie" do
+            expect(response.headers["Set-Cookie"]).
+              to match("domain=.example.com")
+          end
+
+          # There was previously a bug where two cookies would be set
+          # with the same cookie name when using the domain setting
+          # see: https://github.com/thoughtbot/clearance/issues/616
+          it "should not set a second remember token cookie" do
+            cookie_list = response.headers["Set-Cookie"].split("\n")
+            expect(cookie_list.length).to be 1
+          end
+        end
+
+        describe "remember token cookie expiration" do
+          def format_expires(time)
+            if Rails::VERSION::MAJOR > 3
+              time.gmtime.rfc2822
+            else
+              Rack::Utils.rfc2822(time)
+            end
+          end
+
+          context "default configuration" do
+            before do
+              post :create, session: {
+                email: @user.email,
+                password: @user.password
+              }
+            end
+
+            it "is set to 1 year from now" do
+              expect(response.headers["Set-Cookie"]).
+                to match /expires=#{format_expires(1.year.from_now)}/
+            end
+          end
+
+          context "configured with lambda taking one argument" do
+            before do
+              expiration = ->(cookies) { 12.hours.from_now }
```

Comment:
Unused block argument - `cookies`. If it's necessary, use `_` or `_cookies` as an argument name to indicate that it won't be used. Also consider using a proc without arguments instead of a lambda if you want it to accept any arguments but don't care about them.